src/IgnoreList.hxx

```diff
@@ -0,0 +1,28 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+// Copyright The Music Player Daemon Project
+
+#ifndef IGNORE_LIST_HXX
+#define IGNORE_LIST_HXX
+
+#include <string>
+#include <optional>
+#include <vector>
+
+#include "Record.hxx"
+
+struct IgnoreListEntry {
+	std::optional<std::string> artist;
```

Comment:
The config file uses an empty string for "match everything". Why wrap this in `std::optional` here? All this does is add overhead for nothing.

Reply:
That's true. I added optional only to express the semantics in the code. Happy to reduce the overhead though and just go with the strings. Will fix.